bcos-gateway/bcos-gateway/libp2p/P2PMessage.cpp

```diff
@@ -275,8 +315,34 @@ ssize_t P2PMessage::decode(bytesConstRef _buffer)
     uint32_t length = _buffer.size();
     CHECK_OFFSET_WITH_THROW_EXCEPTION(m_length, length);
     auto data = _buffer.getCroppedData(offset, m_length - offset);
-    // payload
-    m_payload = std::make_shared<bytes>(data.begin(), data.end());
+    // raw data cropped from buffer, maybe be compressed or not
+    auto rawData = std::make_shared<bytes>(data.begin(), data.end());
+
+    // uncompress payload
+    // payload has been compressed
+    if ((m_ext & bcos::protocol::MessageExtFieldFlag::Compress) ==
+        bcos::protocol::MessageExtFieldFlag::Compress)
+    {
+        size_t uncompressSize = ZstdCompress::uncompress(ref(*rawData), *m_payload);
+        if (uncompressSize < 1)
```

Comment:
Use micro `UncompressError` to instead of `<1`. 
eg.
`if(uncompressSize == UncompressError)`

Reply:
fixed